ts/client/services/events/genesisBatch.ts

```diff
@@ -0,0 +1,124 @@
+import { BigNumber, Event } from "ethers";
+import { Rollup } from "../../../../types/ethers-contracts/Rollup";
+import {
+    GenesisBatchAlreadySynced,
+    GenesisBatchCommitmentRootMismatch,
+    NotFirstBatch,
+    TooManyGenesisBatchesFound
+} from "../../../exceptions";
+import { Usage } from "../../../interfaces";
+import { CoreAPI, SyncedPoint } from "../../coreAPI";
+import { ConcreteBatch } from "../../features/base";
+import { GenesisCommitment } from "../../features/genesis";
+import { BatchStorage } from "../../storageEngine/batches/interfaces";
+import { ContractEventSyncer } from "./contractEventSyncer";
+
+const genesisBatchID = 0;
+
+/**
+ * Syncs genesis newBatch events from the rollup contract
+ */
+export class GenesisBatchEventSyncer extends ContractEventSyncer {
+    private readonly genesisStateRoot: string;
+    private readonly rollup: Rollup;
+    private readonly syncpoint: SyncedPoint;
+    private readonly batchStorage: BatchStorage;
+
+    constructor(api: CoreAPI) {
+        super(
+            api.contracts.rollup,
+            api.contracts.rollup.filters.NewBatch(null, null, null)
```

Comment:
@ChihChengLiang Would have loved it if the 3rd filter param `batchType` was indexed, would makes this syncer bring in almost no other events. I assume it is not indexed to save gas?